sdk/servicebus/azure-servicebus/azure/servicebus/exceptions.py

```diff
@@ -65,10 +65,25 @@ def _error_handler(error):
 def _create_servicebus_exception(logger, exception, handler):
     error_need_close_handler = True
     error_need_raise = False
-
-    if isinstance(exception, MessageSendFailed):
-        logger.info("Message send error (%r)", exception)
-        error = exception
+    if isinstance(
+        exception,
+        (
+            errors.MessageAccepted,
+            errors.MessageAlreadySettled,
+            errors.MessageModified,
+            errors.MessageRejected,
+            errors.MessageReleased,
+            errors.MessageContentTooLarge,
+            # TODO: which ones should be included. Do we need to separate MessageError from MessageSendFailed?
+        ),
+    ):
+        logger.info("Message error (%r)", exception)
+        error = MessageError(exception)
+        raise error
+    elif isinstance(exception, errors.MessageException):
+        logger.info("Message send failed (%r)", exception)
+        error = MessageSendFailed(exception)
```

Comment:
Wider question that your changes prompted in my head:  Should our top level exceptions be "ServiceBus*" (e.g. ServiceBusSendFailed)?  May need to sync with Anna about what our principles are surrounding this.

I'm also wondering if any of the exceptions in azure-core would suffice as base types, (seperate from the above question, just putting this here as well since it seems apt); per https://azure.github.io/azure-sdk/python_implementation.html#error-handling

Reply:
I was also thinking of prefixing service bus errors with ServiceBus*. But I think MessageSendFailed (or MessageSendError) is concise and easy to understand.
Let's sync offline.